scheduler/core/calculations/targetinfo.py

```diff
@@ -64,6 +65,64 @@ class TargetInfo:
     def mean_airmass(self, interval: npt.NDArray[int]):
         return np.mean(self.airmass[interval])
 
+    @staticmethod
+    def from_dict(ti_dict: Dict) -> 'TargetInfo':
+        return TargetInfo(coord=SkyCoord(ra=np.array(ti_dict['coord']['ra']) * u.deg,
+                                         dec=np.array(ti_dict['coord']['dec']) * u.deg,
+                                         frame=ti_dict['coord']['frame']),
+                          alt=Angle(np.array(ti_dict['alt']['value']), unit=ti_dict['alt']['unit']),
+                          az=Angle(np.array(ti_dict['az']['value']), unit=ti_dict['az']['unit']),
+                          par_ang=Angle(np.array(ti_dict['par_ang']['value']), unit=ti_dict['par_ang']['unit']),
+                          hourangle=Angle(np.array(ti_dict['hourangle']['value']), unit=ti_dict['hourangle']['unit']),
+                          airmass=np.array(ti_dict['airmass']),
+                          sky_brightness=np.array(ti_dict['sky_brightness']),
+                          visibility_slot_idx=np.array(ti_dict['visibility_slot_idx']),
+                          visibility_slot_filter=np.array(ti_dict['visibility_slot_filter']),
+                          visibility_time=TimeDelta(ti_dict['visibility_time']['value'],
+                                                    format=ti_dict['visibility_time']['format']),
+                          rem_visibility_time=TimeDelta(ti_dict['rem_visibility_time']['value'],
+                                                        format=ti_dict['rem_visibility_time']['format']),
+                          rem_visibility_frac=ti_dict['rem_visibility_frac']
+                          )
```

Comment:
These ones, too... I think we can figure out `rem_visibility_frac` from `visibility_time` and `rem_visibility_time`, but if we can store it all, there's no reason not to.

One thing that would be useful is to go through the code and see what fields aren't being used and remove them from dataclasses in `TargetInfo`, `Selection`, `NightEvents`, `NightConfiguration`, etc.